server/src/main/java/org/opensearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java

```diff
@@ -613,7 +614,7 @@ public RestoreSnapshotRequest source(Map<String, Object> source) {
                     throw new IllegalArgumentException("malformed storage_type");
                 }
 
-            } else if (name.equals("source_remote_store_repository")) {
+            } else if (FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE) && name.equals("source_remote_store_repository")) {
```

Comment:
If feature flag is enabled then we should parse the parameter as below. When feature flag is disabled then we should throw proper exception like `"Unsupported parameter " + name + ". Please enable feature flag (flag-name) for this experimental feature"`

Reply:
Please handle it [here](https://github.com/opensearch-project/OpenSearch/pull/8594/files#diff-044b26faac340ba6801867e1c8a943fa978cc797c6d3a354ddc6f30f64702030R668) in `toXContext` as well